### ISSUE TITLE
UEF T2 Field Engineer

### ISSUE THREAD

**Issue by PurpleMossClump** [LOUD-Project/Git-LOUD#83]
The UEF T2 Field Engineer 'xel0209' cannot build T2 shield generators.
The other factions T2 Field Engineers can build T2 shield generators

**Comment by Sprouto** [comment]
Sparky was the original FE - and was heavily modified to turn him into what he is.   He was given a shield, jamming, more HP, and at one point, even a weapon.  Unfortunately, it was never a popular unit - until the edits we made finally got some interest.

The other FE are all products from BrewLAN, and are distinctly different from Sparky - and were introduced quite some time later, to LOUD.   No attempt was ever made to harmonize their actions simply because the BL FE's have special weapons, which are not available to any other engineers.   I did not wish to smear the issue by cross pollination of unique ability between the two. 

Aside from a build list change, which is easy enough to effect - the biggest issue surrounds the mixed use of engineers.   The AI currently doesn't build FE because they are not, presently, distinguishable from standard engineers.   The AI would be incapable of building any 'special' structures, and would expect that the FE could be everything the standard engineers, which is not necessarily true.

The proper solution would be to create the data necessary for the AI to tell the difference, and then have builders to construct the FE, and a full series of conditions, either to keep FE's from taking on tasks they cannot do - or standard engineers from taking on FE tasks.   In other words, a complete separation of tasks and responsibility - as is done with the ACU and SACU - compared to standard engineers.

I'm not against this kind of work - by any means - but it is considerable.   While a 'hack' has been proposed to allow the AI to build FE and use them in all tasks, I don't feel that's a suitable solution.   There would still be no builder to construct FE's unless they are just 'slotted in' as a standard replacement to other engineers - on a % basis, in which case, yes, you'll see them built - but the AI would still be trying to employ them as standard engineers - with standard tasks.   That's no real gain for the AI.
